components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryRecoveryDataStore.java

```diff
@@ -241,16 +232,35 @@ public void invalidate(String userId, int tenantId) throws IdentityException {
     }
 
     private void deleteOldResourcesIfFound(Registry registry, String userName, String secretKeyPath) {
+
+        Collection collection = null;
+        try {
+            collection = (Collection) registry.get(secretKeyPath.toLowerCase());
+        } catch (RegistryException e) {
+            log.error("Error while deleting the old confirmation code. Unable to find data collection in registry." + e);
```

Comment:
The best approach would be to pass the exception object to the error(message, e) method without concatenating it to the message.